.circleci/config.yml

```diff
@@ -930,8 +930,8 @@ jobs:
       module:
         description: Go Module Name
         type: string
-    docker:
-      - image: <<pipeline.parameters.ci_builder_image>> # only used to enable codecov.
+    machine:
+      image: <<pipeline.parameters.base_image>>
```

Comment:
This feels wrong. We're not controlling the environment we test in - the default base_image is just "default". I'd expect to be using the CI builder for running tests to be honest. Given neither the unit tests or fuzz tests are on the critical path for CI completion (e2e tests take way longer) I'm not sure it's worth switching to a machine executor here.

Reply:
CI builder is expensive and not actually necessary for running these tests -- could we just pin this to another image instead?

Reply:
Could we just do something like this instead?

```
docker:
      - image: cimg/go:1.21
```

Reply:
The version of go we test on would get out of sync with what we actually build on. For things like cannon/op-program that can cause major issues to go unnoticed. We could do the work to make ci-builder more layered so we have a basic go version and then layer in things like forge etc on top of that, allowing us to pull smaller images (or we could just use a hosting provider that's not so expensive...). But I don't think we should degrade our testing to cut costs - it will come back to bite us.

Reply:
Alternatively if we want to control the environment we could create a lighter ci builder to avoid some of the export cost

Reply:
For now I'm reverting this change, we can discuss later